### ISSUE TITLE
Ignored user-provided command-line options

### ISSUE THREAD

**Issue by mscuttari** [OpenModelica/OpenModelica#10533]
### Description

`__OpenModelica_commandLineOptions` overrides user-provided command line options

### Steps to Reproduce

Consider the following model:
```modelica
model Test
  Real x;
equation
  der(x) = -x + 1;
  annotation(__OpenModelica_commandLineOptions= "-d=initialization --daeMode");
end Test;
```
And run the following command through CLI:
`omc Test.mo -i=Test -f`
The flattened model is not printed and the C files are generated.

After some tests, the `--daeMode` flag turned out to be the culprit. After removing it, the flattened model gets printed. However, I don't exclude the problem to be wider.

### Expected Behavior

The user-provided flags should be merged with the ones provided by the source file.

### Version and OS

 - OpenModelica Version: 1.22.0~dev-16-g8afd6ca
 - OS: Ubuntu 20.04

**Comment by casella** [comment]
@perost, @mahge I'm not 100% sure this is a regression, but I understand this may have been working in the past, then we tried to be "smarter" when merging command-line flags from the annotations and command-line flags from the API, probably neglecting the fact that command line flags can, well, come from the command line 😅 

This feature is useful if you use OMC as a flattening front-end, which is then most easily called from the command line, rather than through a script, e.g. as we do with our MARCO complier. 

Do you mind having a look? Thanks!

**Comment by mahge** [comment]
This was an issue with the `--daeMode` option caused by the  changes in #10418. The option was left in a state where it would always run the backend regardless of `-s` flag. In addition, it never printed any flat Modelica string. So if you removed the annotations but specify --daeMode on the command line interface, you would have gotten the same behavior. 

```mo
model Test
  Real x;
equation
  der(x) = -x + 1;
end Test;
```

with daeMode applied on the CLI
```sh
omc Test.mo -i=Test -f --daeMode
```

behaved the same way.

It should be fixed by #10552.

**Comment by casella** [comment]
Great, thanks!

@mscuttari, I already updated our server with the latest nightly build containing the fix, please try that out.
